src/deepsparse/loggers/prometheus_logger.py

```diff
@@ -0,0 +1,188 @@
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import os
+from collections import defaultdict
+from typing import Any, Optional
+
+from deepsparse.loggers.base_logger import BaseLogger
+from prometheus_client import REGISTRY, Histogram, start_http_server, write_to_textfile
+
+
+__all__ = ["PrometheusLogger"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class PrometheusLogger(BaseLogger):
+    """
+    Logger that uses the official Prometheus python client
+    (https://github.com/prometheus/client_python) to monitor the
+    inference pipeline
+
+    :param port: the port used by the client. Default is 8000
+    :param text_log_save_dir: the directory where the text log files
+        are saved. By default, the python working directory
+    :param text_log_save_freq: the frequency of saving the text log
+        files. E.g. if `text_log_save_freq` = 10, text logs are dumped
+        after every tenth forward pass (the number of forward passes
+        equals the number of times `log_latency` method is called).
+    :param text_log_file_name: the name of the text log file.
+        Default: `prometheus_logs.prom`.
+    """
+
+    def __init__(
+        self,
+        port: int = 8000,
+        text_log_save_dir: str = os.getcwd(),
+        text_log_save_freq: int = 10,
+        text_log_file_name: Optional[str] = None,
+    ):
+        self.port = port
+        self.text_log_save_freq = text_log_save_freq
+        self.text_log_save_dir = text_log_save_dir
+        self.text_log_file_name = text_log_file_name or f"{self.identifier}_logs.prom"
+        self._setup_client()
+
+        # the data structure responsible for the instrumentation
+        # of the metrics
+        self.metrics = defaultdict(lambda: defaultdict(str))
+        # internal counter tracking the number of calls per each
+        # pipeline i.e. self._counter: Dict[str, int]
+        self._counter = {}
+
+        super().__init__()
+
+    @property
+    def identifier(self) -> str:
+        return "prometheus"
+
+    @property
+    def text_logs_path(self) -> str:
+        """
+        :return: The path to the text file with logs
+        """
+        return os.path.join(self.text_log_save_dir, self.text_log_file_name)
+
+    @property
+    def counter(self) -> int:
+        """
+        While self._counter collects the number of forward passes per
+        pipeline inference, self.counter returns a sum of all the calls
+
+        :return: The total amount of times the logger has been called by
+            any pipeline
+        """
+        return sum(self._counter.values())
```

Comment:
Is `self._counter` only used here to get the number of times logger has been called? Can self._counter just be a single value instead of a dictionary?

Reply:
self._counter counts how many inferences were done per pipeline
self.counter sums them up and returns the total count of interences of the logger

Reply:
It looks like self._counter[pipeline_name] is incremented once per log_latency call - can you just have an integer attribute that you increment instead of the dictionary? Then you could just use `self._counter` directly instead of doing the sum of values